src/newton1d.jl

```diff
@@ -0,0 +1,74 @@
+function newton1d{T}(f::Function, f_prime::Function, x::Interval{T};
+                    epsX=eps(T), epsf=eps(T), debug=false)
+
+    L = Interval[]
+    R = Root[]
+    z = zero(x.lo)
+    push!(L, x)
+    while !isempty(L)
+        X = pop!(L)
+        mid_x = mid(X)
+        if !(z in f_prime(X))
+            # O ∉ f'(X)
+            while true
+                mid_x = mid(X)
+                N = mid_x - (f(Interval(mid_x))/f_prime(X))
+                X = X ∩ N
+                if isempty(X)
+                    break
+                elseif z in f(Interval(prevfloat(mid_x), nextfloat(mid_x))) # What if the root is one of the end points of the interval. That way, this condition will never satisfy even thoughthe interval becomes very small. Explicit check for that?
```

Comment:
@dpsanders 
What if the root is one of the end points of the interval. That way, this condition will never satisfy even though the interval becomes very small. Should there be an explicit check for that?

Reply:
I'm not sure what you mean?